### PR TITLE
[REF] further simplification of custom field create action

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Code cleanup towards https://github.com/civicrm/civicrm-core/pull/14694

Before
----------------------------------------
Less readable

After
----------------------------------------
More readable

Technical Details
----------------------------------------
Test cover is pretty good on this function - specifically testGetFieldsForImport checks that the custom field id is still appended and testCreateCustomFieldColumnName tests that a column_name can be passed in & is then respected

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Core/BAO/CustomField.php [hunk 1]
```diff
@@ -153,38 +153,24 @@ public static function dataToHtml() {
    */
   public static function create($params) {
     $transaction = new CRM_Core_Transaction();
-    $op = empty($params['id']) ? 'create' : 'edit';
-    $origParams = array_merge([], $params);
-    $params = self::prepareCreate($params, $op);
+    $isPredeterminedColumnName = isset($params['column_name']);
```
Comment (eileenmcnaughton) [inline CRM/Core/BAO/CustomField.php]: origParams really only serve to determine if the column name was preset

**Comment by pradpnayak** [comment]
I guess this needs rebasing

**Comment by eileenmcnaughton** [comment]
rebased

File: CRM/Core/BAO/CustomField.php [hunk 2]
```diff
@@ -153,42 +153,31 @@ public static function dataToHtml() {
...
     $transaction = new CRM_Core_Transaction();
-    $origParams = array_merge([], $params);
     $params = self::prepareCreate($params);
 
     $customField = new CRM_Core_DAO_CustomField();
     $customField->copyValues($params);
     $customField->save();
 
-    // make sure all values are present in the object for further processing
-    $customField->find(TRUE);
```
Comment (colemanw) [inline CRM/Core/BAO/CustomField.php]: I remember there was some reason we had to add this...

Comment (eileenmcnaughton) [reply CRM/Core/BAO/CustomField.php]: It's still there - just moved

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by colemanw** [comment]
@eileenmcnaughton needs rebase

**Comment by eileenmcnaughton** [comment]
Ok -another one got merged that incorporateed this - closing - will figure out the 'next in line'

**Pull request closed without merging**